src/main/java/org/jenkinsci/plugins/p4/PerforceScm.java

```diff
@@ -220,8 +224,22 @@ public String getKey() {
 			logger.fine("No credential for perforce");
 			return null;
 		}
+
+		// Retrieve item from request
+		StaplerRequest req = Stapler.getCurrentRequest();
+		Job job = req == null ? null : req.findAncestorObject(Job.class);
```

Comment:
And why are you doing this calculation now? Do you actually need to contact the server in order to determine whether to use `SwarmBrowser` (whatever that is) or something else? (And what would that something else be?)

`guessBrowser` is just intended as a heuristic, for SCMs which multiple kinds of web-based browsers depending on hosting service, etc. That is probably not the case for P4.